client/my-sites/google-my-business/connect-button/index.js

```diff
@@ -0,0 +1,135 @@
+	createOrUpdateConnection = () => {};
+
+	// override `removeConnection` to remove the keyring connection instead of the publicize one
+	removeConnection = () => {
+		this.setState( { isDisconnecting: true } );
+		this.props.deleteStoredKeyringConnection( last( this.props.keyringConnections ) );
+	};
+
+	onClick = () => {
+		this.performAction();
+		this.props.onClick();
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( this.props.availableExternalAccounts, nextProps.availableExternalAccounts ) ) {
+			this.setState( {
+				isConnecting: false,
+				isDisconnecting: false,
+			} );
+		}
+
+		if ( ! this.state.isAwaitingConnections ) {
+			return;
+		}
+
+		this.setState( {
+			isAwaitingConnections: false,
+			isRefreshing: false,
+		} );
+
+		if ( this.didKeyringConnectionSucceed( nextProps.availableExternalAccounts ) ) {
+			this.setState( { isConnecting: false } );
+			this.props.onConnect();
+		}
+	}
+
+	render() {
+		const { service, translate } = this.props;
+		const { isConnecting, isRefreshing, isDisconnecting } = this.state;
+		const status = this.getConnectionStatus( service.ID );
+		let primary = false,
+			warning = false,
+			label;
+
+		const isPending = 'unknown' === status || isDisconnecting || isRefreshing || isConnecting;
+
+		if ( 'unknown' === status ) {
+			label = translate( 'Loading…' );
+		} else if ( isDisconnecting ) {
+			label = translate( 'Disconnecting…' );
+		} else if ( isRefreshing ) {
+			label = translate( 'Reconnecting…' );
+			warning = true;
+		} else if ( isConnecting ) {
+			label = translate( 'Connecting…' );
+		} else if ( 'connected' === status ) {
+			label = translate( 'Disconnect' );
+		} else {
+			label = this.props.children;
+			primary = true;
+		}
+
+		return (
+			<Button primary={ primary } scary={ warning } onClick={ this.onClick } disabled={ isPending }>
+				{ label }
+			</Button>
+		);
+	}
+}
+
+// `connectFor` HOC requires the `service` props to be passed
+// Therefore we need to wrap the resulting component into a parent component
+const GoogleMyBusinessConnectButtonConnected = connectFor(
+	GoogleMyBusinessConnectButton,
+	( state, props ) => {
+		return {
+			...props,
+			removableConnections: props.keyringConnections,
+			fetchConnection: props.requestKeyringConnections,
+			service: getKeyringServiceByName( state, 'google_my_business' ),
+			siteUserConnections: props.keyringConnections.map( connection => ( {
+				...connection,
+				keyring_connection_ID: connection.ID,
+			} ) ),
+		};
+	},
+	{
+		deleteStoredKeyringConnection,
+	}
+);
+
+export default connect( state => ( {
+	service: getKeyringServiceByName( state, 'google_my_business' ),
+} ) )( props => (
+	<div>
+		<QueryKeyringServices />
+		{ props.service && <GoogleMyBusinessConnectButtonConnected { ...props } /> }
+		{ ! props.service && <Button disabled={ true }>{ props.children }</Button> }
```

Comment:
Why here we use `children` but `GoogleMyBusinessConnectButton` uses self-generated `label` in `render()` ?

Reply:
Ohh, I see, my bad, it's used there as well. sorry.